sormas-stats-sitrep/utils/caseExport.R

```diff
@@ -0,0 +1,190 @@
+  
+  # District table SQL query
+  queryDistrict <- paste0("SELECT DISTINCT id AS district_id,
+    name AS district_name
+    
+    FROM public.district
+    
+    WHERE archived = FALSE
+                          ")
+  districts <- dbGetQuery(sormas_db, queryDistrict)
+  
+  # Region table SQL query
+  queryRegions <- paste0("SELECT DISTINCT id AS region_id,
+    name AS region_name
+    
+    FROM public.region 
+    
+    WHERE archived = FALSE
+                         ")
+  regions <- dbGetQuery(sormas_db, queryRegions)
+  
+  # Populationdata table SQL query 
+  queryPopulation <- paste0("SELECT DISTINCT id AS population_id,
+                            district_id,
+                            region_id, 
+                            sex AS sex_population,
+                            agegroup AS agegroup_population,
+                            population
+                            
+                            FROM public.populationdata
+                            ")
+  population_data <- dbGetQuery(sormas_db, queryPopulation)
+  
+  # Merging population data with district names
+  population_data <- population_data %>% 
+    dplyr::left_join(., districts, by = 'district_id' )
+  
+  # Merging data from cases, persons, hospitalizations, districts and regions 
+  # exports into one line listed DataFrame with every row being a case 
+  line_list_cases <- cases %>% 
+    dplyr::left_join(., persons, by = 'person_id') %>% 
+    dplyr::left_join(., symptoms, by = 'symptoms_id') %>% 
+    dplyr::left_join(., hospitalizations, by = 'hospitalization_id') %>% 
+    dplyr::left_join(., districts, by = 'district_id') %>% 
+    dplyr::left_join(., regions, by = 'region_id') %>% 
+    dplyr::mutate(report_date_case = as.Date(format(report_date_case, "%Y-%m-%d")),
+                  admission_date = as.Date(format(admission_date, "%Y-%m-%d")),
+                  onset_date = as.Date(format(onset_date, "%Y-%m-%d"))) 
+    
+  
+  ## Provisory geo shapes export:
+  # geo_shapes data needs to be placed in data/geo folder and table key needs 
+  # to be placed in data/key folder.
+  
+  # Loading geoshapes from folder, not part of SORMAS data for now
+  # data source: https://gadm.org/download_country.html
+  files_geo <- list.files('data/geo', full.names = TRUE)
+  
+  # Reading in geo_shapes data
+  geo_shapes <- readRDS(files_geo) %>%
+    dplyr::select(NAME_2, geometry) %>% 
+    dplyr::rename(district_name_geo = NAME_2)
+  
+  ## Merging Population and geo data using a table key
+  
+  # Loading key table from folder
+  files_key <- list.files('data/key', full.names = TRUE)
+  
+  # Reading in key table
+  key <- readxl::read_xlsx(files_key)
+  
+  # Merging population data and geo_shapes data
+  population_geo_data <- population_data %>%
+    dplyr::left_join(., key, by = 'district_name') %>% # joining w/ the matching key
+    dplyr::left_join(., geo_shapes, by = 'district_name_geo') # joining w/ the geo_shapes
```

Comment:
Only things that are part of sormas database should be here, thus we can move this to its own function